internal/controllers/provisioningrequest_clusterinstall.go

```diff
@@ -447,3 +419,237 @@ func (t *provisioningRequestReconcilerTask) updateClusterProvisionStatus(ci *sit
+		utils.IsClusterProvisionCompleted(t.object) {
+		err = addSuppressedInstallManifests(renderedCI)
+		if err != nil {
+			return fmt.Errorf(
+				"failed to add suppressed install manifests to the rendered ClusterInstance (%s): %w",
+				renderedCI.GetName(), err)
+		}
+		disallowedChanges = slices.DeleteFunc(disallowedChanges, func(field string) bool {
+			return field == "clusterImageSetNameRef"
+		})
+	}
+	if existingCISuppressed, ok := existingCI.Object["spec"].(map[string]any)["suppressedManifests"].([]any); ok {
+		renderedCISuppressed, ok := renderedCI.Object["spec"].(map[string]any)["suppressedManifests"].([]any)
+		if !ok {
+			renderedCISuppressed = []any{}
+		}
+		for _, suppressedManifest := range existingCISuppressed {
+			if !slices.ContainsFunc(renderedCISuppressed, func(item any) bool {
+				return item.(string) == suppressedManifest.(string)
+			}) {
+				renderedCISuppressed = append(renderedCISuppressed, suppressedManifest)
+			}
+		}
+		renderedCI.Object["spec"].(map[string]any)["suppressedManifests"] = renderedCISuppressed
+	}
+
+	if len(disallowedChanges) > 0 &&
+		utils.IsClusterProvisionCompleted(t.object) {
+		return utils.NewInputError(
+			"detected disallowed changes in immutable fields: %s", strings.Join(disallowedChanges, ", "))
+	}
+
+	disallowedChanges = append(disallowedChanges, scalingNodes...)
+	if len(disallowedChanges) > 0 &&
+		utils.IsClusterProvisionInProgress(t.object) {
+		return utils.NewInputError(
+			"updates to ClusterInstance parameters are disallowed during cluster installation, "+
+				"detected changes in fields: %s", strings.Join(disallowedChanges, ", "))
+	}
+
+	return nil
+}
+
+// extractNodeDetails extracts necessary node details from the existing ClusterInstance
+func extractNodeDetails(existingCI *unstructured.Unstructured) (map[string]nodeInfo, error) {
+	nodes := existingCI.Object["spec"].(map[string]any)["nodes"].([]any)
+	extractedNodesInfo := make(map[string]nodeInfo) // keyed by hostname
+
+	for _, node := range nodes {
+		nodeMap := node.(map[string]any)
+		hostname, ok := nodeMap["hostName"].(string)
+		if !ok {
+			return nil, fmt.Errorf("valid hostName not found")
+		}
+
+		extractedNodeInfo := nodeInfo{}
+		v, ok := nodeMap["bmcAddress"].(string)
+		if !ok {
+			return nil, fmt.Errorf("valid bmcAddress not found")
+		}
+		extractedNodeInfo.bmcAddress = v
+
+		v, ok = nodeMap["bootMACAddress"].(string)
+		if !ok {
+			return nil, fmt.Errorf("valid bootMACAddress not found")
+		}
+		extractedNodeInfo.bootMACAddress = v
+
+		bmcCreds, ok := nodeMap["bmcCredentialsName"].(map[string]any)
+		if !ok {
+			return nil, fmt.Errorf("valid bmcCredentialsName not found")
+		}
+		v, ok = bmcCreds["name"].(string)
+		if !ok {
+			return nil, fmt.Errorf("valid name not found in bmcCredentialsName")
+		}
+		extractedNodeInfo.bmcCredentialsName = v
+
+		// Extract interface macAddress by interface name
+		nodeNetwork, ok := nodeMap["nodeNetwork"].(map[string]any)
```

Comment:
Maybe not worth returning errors on fields that are not required in the ClusterInsance CRD.